// wasm-onlyoffice-sdk/dist/react/index.mjs
import { jsx as m } from "react/jsx-runtime";
import { useRef as d, useEffect as z, useLayoutEffect as B } from "react";
import { EditorServer as V, getDocumentType as _, MockSocket as f, createXHRProxy as G, io as J } from "../index.mjs";
function Z({
  assetsPath: H,
  x2tPath: I = "/x2t-1",
  file: w,
  fileUrl: y,
  newDocument: g,
  language: P = "en",
  theme: W = "theme-light",
  user: E = { id: "uid", name: "User" },
  onReady: D,
  onDocumentStateChange: v,
  onSave: k,
  onError: L,
  style: X,
  className: j
}) {
  const l = d(!1), b = H.replace(/\/$/, ""), a = d(D), u = d(v), p = d(k), r = d(L);
  a.current = D, u.current = v, p.current = k, r.current = L, z(() => {
    const c = (n) => {
      l.current && (n.preventDefault(), n.returnValue = "");
    };
    return window.addEventListener("beforeunload", c), () => window.removeEventListener("beforeunload", c);
  }, []), B(() => {
    const c = b + "/web-apps/apps/api/documents/api.js", n = new V({ x2tPath: I, user: E, onSave: (e, t) => {
      var o;
      return (o = p.current) == null ? void 0 : o.call(p, e, t);
    } });
    w ? n.open(w) : y ? n.openUrl(y) : g ? n.openNew(g) : n.openNew("docx");
    const i = n.getDocument(), M = _(i.fileType);
    let s = null;
    const T = ({ socket: e }) => n.handleConnect({ socket: e }), q = ({ socket: e }) => n.handleDisconnect({ socket: e });
    f.on("connect", T), f.on("disconnect", q);
    const O = () => {
      var U, C;
      const e = document.querySelector('iframe[name="frameEditor"]'), t = e == null ? void 0 : e.contentWindow, o = e == null ? void 0 : e.contentDocument;
      if (!o || !t) {
        (U = r.current) == null || U.call(r, new Error("Iframe not loaded"));
        return;
      }
      const A = G(t.XMLHttpRequest), N = t.Worker;
      A.use((h) => n.handleRequest(h)), Object.assign(t, {
        io: J,
        XMLHttpRequest: A,
        Worker: function(h, $) {
          return new N(new URL(h, location.origin).href, $);
        }
      });
      const S = o.createElement("script");
      S.src = new URL(c, location.origin).href, o.body.appendChild(S), (C = a.current) == null || C.call(a);
    }, x = () => {
      s = new window.DocsAPI.DocEditor("placeholder", {
        isLocalFile: !0,
        document: {
          fileType: i.fileType,
          key: i.key,
          title: i.title,
          url: i.url,
          permissions: {
            edit: i.fileType !== "pdf",
            chat: !1,
            rename: !0,
            protect: !0,
            review: !1,
            print: !1
          }
        },
        documentType: M,
        editorConfig: {
          lang: P,
          coEditing: { mode: "fast", change: !1 },
          user: { ...E },
          customization: {
            uiTheme: W,
            features: { spellcheck: { change: !1 } }
          }
        },
        events: {
          onAppReady: () => O(),
          onDocumentStateChange: (e) => {
            var t;
            e.data && (l.current = !0), (t = u.current) == null || t.call(u, e.data);
          },
          onError: (e) => {
            var t;
            return (t = r.current) == null ? void 0 : t.call(r, new Error(String(e)));
          },
          onSaveDocument: () => {
            l.current = !1;
          },
          writeFile: () => {
            l.current = !1;
          }
        },
        width: "100%",
        height: "100%"
      });
    };
    return (() => {
      var t;
      if ((t = window.DocsAPI) != null && t.DocEditor) {
        x();
        return;
      }
      let e = document.querySelector(`script[src="${c}"]`);
      e || (e = document.createElement("script"), e.src = c, document.head.appendChild(e)), e.onload = () => x(), e.onerror = () => {
        var o;
        return (o = r.current) == null ? void 0 : o.call(r, new Error("Failed to load DocsAPI script"));
      };
    })(), () => {
      var e;
      f.off("connect", T), f.off("disconnect", q), (e = s == null ? void 0 : s.destroyEditor) == null || e.call(s), n.destroy();
    };
  }, []);
  const F = b + "/web-apps/apps/api/documents/preload.html";
  return /* @__PURE__ */ m("div", { style: { width: "100%", height: "100%", ...X }, className: j, children: /* @__PURE__ */ m("div", { id: "placeholder", style: { width: "100%", height: "100%" }, children: /* @__PURE__ */ m(
    "iframe",
    {
      style: { width: 0, height: 0, display: "none" },
      src: F
    }
  ) }) });
}
export {
  Z as OnlyOfficeEditor
};
